plugins/Marketplace/config/config.php

```diff
@@ -1,16 +1,17 @@
 <?php
 
 use Piwik\Config\GeneralConfig;
+use Piwik\Http;
 use Piwik\Plugins\Marketplace\Api\Service;
 use Piwik\Plugins\Marketplace\LicenseKey;
 use Psr\Container\ContainerInterface;
 
 return array(
     'MarketplaceEndpoint' => function (ContainerInterface $c) {
-        $domain = 'https://plugins.matomo.org';
+        $domain = 'http://plugins.matomo.org';
 
-        if (GeneralConfig::getConfigValue('force_matomo_ssl_request') === 0) {
-            $domain = str_replace('https://', 'http://', $domain);
+        if (Http::isUpdatingOverHttps() && GeneralConfig::getConfigValue('force_matomo_http_request') != 1) {
+            $domain = str_replace('http://', 'https://', $domain);
```

Comment:
also is this defaulting to https? We don't want to default to https just yet, right?